monai/inferers/merger.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+
+import torch
+
+from monai.utils import ensure_tuple_size
+
+__all__ = ["Merger", "AvgMerger"]
+
+
+class Merger(ABC):
```

Comment:
It's not clear from the docstrings how one would go about implementing a merger class given the methods that need to be implemented. It would need to be explicit here what each method does in more detail and how they get called together. If possible it would be better to provide a default implementation of the methods given so that one need implement only those methods that are totally necessary.

Reply:
I updated the based class and also added more description on what needs to be implemented in concrete classes.